RPT_Application/src/main/java/com/harman/user_database/EmployeeService.java

```diff
@@ -0,0 +1,68 @@
+package com.harman.user_database;
+
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class EmployeeService {
+
+	@Autowired
+	private EmployeeRepository employeeRepository;
+
+	@Autowired
+	private BCryptPasswordEncoder passwordEncoder;
+
+	private static final Logger LOGGER = Logger.getLogger(EmployeeService.class.getName());
+
+	private EmployeeService() {
+	}
+
+	public List<Employee> findAll() {
+		return employeeRepository.findAll();
+	}
+
+	public int count() {
+		return (int) employeeRepository.count();
+	}
+
+	public Employee findByUsername(String username) {
+		return employeeRepository.findByUsername(username);
+	}
+
+	public Employee findById(String id) {
+		return employeeRepository.findById(id).orElse(null);
+	}
+
+	public boolean delete(Employee employee) { //false - employee wasn't deleted, true - employee was deleted
+		if (employee == null) {
+			LOGGER.log(Level.SEVERE, "Employee is null. Check why do you delete null Employee.");
+			return false;
+		}
+		if (employee.getAuthority() == "ROLE_ADMIN") {
+			LOGGER.log(Level.SEVERE, "Employee is admin. No one is allowed to remove admin.");
+			return false;
+		}
+		employeeRepository.delete(employee);
```

Comment:
What if this will return false?

Reply:
employeeRepository.delete(employee) will return false? Or public boolean delete(Employee employee) will return false?

employeeRepository.delete() is void type

and if delete(Employee employee) will return false then this happens in EmployeesCRUDView:
if (employeeService.delete(employee)) {
				grid.setItems(employeeService.findAll());
				dialog.close();
			} else {
				Dialog notAllowedDialog = new Dialog();
				Label label2 = new Label("You can not remove admin");
				notAllowedDialog.add(label2);
				notAllowedDialog.open();
			}